File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    const splt = s.trim().split(" ");
    const ret = new Array();

    for (let i = splt.length-1; i >= 0; i--) {
        const temp = splt[i]
        if (temp != "") {
            ret.push(splt[i] )
        }
    }

    return String(ret.join(" "))
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] strs = s.trim().split(" ");
        StringBuilder ret = new StringBuilder();

        for (int i=strs.length-1; i >=0; i--) {
            if (!strs[i].isEmpty()){
                ret.append(strs[i]).append(" ");
            }
        }

        return ret.toString().trim();
    }
}